android/src/main/java/com/oney/WebRTCModule/CameraCaptureController.java

```diff
@@ -48,7 +60,132 @@ public CameraCaptureController(Context context, CameraEnumerator cameraEnumerato
+            } catch (Exception e) {
+                Log.d(TAG, "failed to find device with id: " + deviceId);
+            }
+        }
+
+
+        // Otherwise, use facingMode (defaulting to front/user facing).
+        if (cameraName == null) {
+            cameraIndex = -1;
+            for (String name : deviceNames) {
+                cameraIndex++;
+                if (cameraEnumerator.isFrontFacing(name) == isFrontFacing) {
+                    cameraName = name;
+                    break;
+                }
+            }
+        }
+
+        if (cameraName == null) {
+            onFinishedCallback.accept(new Exception("OverconstrainedError: could not find camera with deviceId: " + deviceId + " or facingMode: " + facingMode));
+            return;
+        }
+        
+        final String finalCameraName = cameraName; // For lambda reference
+        boolean shouldSwitchCamera = false;
+        try {
+            int currentCameraIndex = Integer.parseInt(currentDeviceId);
+            shouldSwitchCamera = cameraIndex != currentCameraIndex;
+        } catch (Exception e) {
+            shouldSwitchCamera = true;
+            Log.d(TAG, "Forcing camera switch, couldn't parse current device id: " + currentDeviceId);
+        }
+
+        CameraVideoCapturer capturer = (CameraVideoCapturer) videoCapturer;
+        Runnable changeFormatIfNeededAndFinish = () -> {
+            saveConstraints.run();
+            if (targetWidth != oldTargetWidth ||
+                    targetHeight != oldTargetHeight ||
+                    targetFps != oldTargetFps) {
+                updateActualSize(finalCameraName, videoCapturer);
+                capturer.changeCaptureFormat(targetWidth, targetHeight, targetFps);
+            }
+            if (onFinishedCallback != null) {
+                onFinishedCallback.accept(null);
+            }
+        };
+
+        if (shouldSwitchCamera) {
+            capturer.switchCamera(new CameraVideoCapturer.CameraSwitchHandler() {
+                @Override
+                public void onCameraSwitchDone(boolean isFrontCamera) {
+                    CameraCaptureController.this.isFrontFacing = isFrontCamera;
+                    changeFormatIfNeededAndFinish.run();
+                }
+
+                @Override
+                public void onCameraSwitchError(String s) {
+                    Exception e = new Exception("Error switching camera: " + s);
+                    Log.e(TAG, "OnCameraSwitchError", e);
+                    if(onFinishedCallback != null) {
+                        onFinishedCallback.accept(e);
+                    }
+                }
+            }, cameraName);
+        } else {
+            // No camera switch needed, just change format if needed.
+            changeFormatIfNeededAndFinish.run();
+        }
+    }
+
+    /**
+     * Use applyConstraints instead.
+     * @deprecated
+     */
+    @Deprecated
+    public void switchCamera(Consumer<Exception> onFinishedCallback) {
```

Comment:
Why keep this? Can't we re-implement it in JS using `applyConstraints` internally?